tests/unit/test_application.py

```diff
@@ -274,3 +275,18 @@ def test_application_build_with_extensions(monkeypatch, extension_source, new_di
 
     project = app.get_project()
     assert "fake-extension/fake-part" in project.parts
+
+
+def test_application_managed_core20_fallback(monkeypatch, new_dir, mocker):
+    monkeypatch.setenv("CRAFT_DEBUG", "1")
+    monkeypatch.setenv("SNAPCRAFT_BUILD_ENVIRONMENT", "managed-host")
+
+    (new_dir / "snap").mkdir()
+
+    mock_legacy_run = mocker.patch("snapcraft_legacy.cli.legacy.legacy_run")
+    application.create_app = Mock()
```

Comment:
you need to use `mocker` here too, otherwise this mock is going to outlive the test as these module objects are global

something like
```python
mock_create_app = mocker.patch.object(application, "create_app")

Reply:
^pretty sure that's why the tests are failing actually

Reply:
Ah ok, that's unexpected.  Will fix now.